src/c++/perf_analyzer/data_loader.cc

```diff
@@ -481,22 +479,26 @@ DataLoader::ReadTensorData(
           }
         }
 
-        if (!tensor.HasMember("content")) {
-          return cb::Error(
-              "missing content field. ( Location stream id: " +
-                  std::to_string(stream_index) +
-                  ", step id: " + std::to_string(step_index) + ")",
-              pa::GENERIC_ERROR);
-        }
+        if (tensor.HasMember("b64")) {
+          content = &tensor;
+        } else {
+          if (!tensor.HasMember("content")) {
+            return cb::Error(
+                "missing content field. ( Location stream id: " +
+                    std::to_string(stream_index) +
+                    ", step id: " + std::to_string(step_index) + ")",
+                pa::GENERIC_ERROR);
+          }
 
-        content = &tensor["content"];
+          content = &tensor["content"];
+        }
       }
 
       if (content->IsArray()) {
         RETURN_IF_ERROR(SerializeExplicitTensor(
             *content, io.second.datatype_, &it->second));
       } else {
-        if (content->HasMember("b64")) {
+        if (content->IsObject() && content->HasMember("b64")) {
```

Comment:
This avoids a segfault if the user supplies content that isn't an Array or Object (it falls through to lower code that throws an error)

Reply:
It seems we special case b64 a few times. Would it make more sense to pull these checks into its own if/else block and if not b64, it skips all of it?

Reply:
coverage first and then I'll clean up and refactor